### ISSUE TITLE
Allow custom colors for project names

### ISSUE THREAD

**Issue by sheremet-va** [vitest-dev/vitest#6481]
### Clear and concise description of the problem

It is not possible to configure colors of the project names in UI or in the CLI.

### Suggested solution

Allow accepting the `name` property as an object:

```ts
interface ProjectName {
  name: string
  color: keyof Colors
}
```

We should also change the default colors to be background ones by default (this was initially part of #6329):
<img width="742" alt="Screenshot 2024-08-12 at 23 03 51" src="https://github.com/user-attachments/assets/a3b5d823-c6d8-495d-b6d3-6b637fdb5e96">

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by AriPerkkio** [comment]
Related to https://github.com/vitest-dev/vitest/issues/6338

**Comment by kevinmarrec** [comment]
@sheremet-va Vitest should provide somehow at minimum 2 colorsets (until custom colors for project names come out), following a dark/light theme convention, because when using Vitest with workspaces, when running Vitest either in a terminal with black background or in Github Actions with black background, it's not accessible (especially on Github actions) :

![Image](https://github.com/user-attachments/assets/7cbd6024-4585-46e6-96b6-7259d59d1c5f)

![Image](https://github.com/user-attachments/assets/574ce6c4-80e8-457d-9151-413d2b3900ed)

**Comment by rgrove** [comment]
I was surprised to discover that in Vitest 3, project names are completely unreadable in my terminal with a dark color scheme. The big blank space on the left in the screenshot below is where (invisible) project names are printed. I've obscured the filenames on the right since this is a non-public codebase.

![](https://github.com/user-attachments/assets/2e5e644e-fd44-4065-8c36-8cf05cbdd097)

**Comment by kevinmarrec** [comment]
I think the first color picked is `black` :

https://github.com/vitest-dev/vitest/blob/9acc13f367988ff5711f51d8b360a7f6d6602d2c/packages/vitest/src/node/reporters/renderers/utils.ts#L226

@rgrove  And I guess you're being pretty unlucky because your terminal background is exactly the same `black` color is using (from `tinyrainbow` library), so you don't see it at all 😵

**Comment by rgrove** [comment]
Yeah, that looks like the problem! Seems like this issue with dark backgrounds could be mitigated by removing black from the list of possible colors?
